contributingGuides/NAVIGATION.md

```diff
@@ -70,3 +70,74 @@ Using [react-freeze](https://github.com/software-mansion/react-freeze) allows us
 
 - To keep the navigation state that was present before changing the layout, we save the state on every change and use it for the `initialState` prop.
 Changing the layout means that every component inside `NavigationContainer` is mounted anew.
+
+## Why we need to use minimal action in the `linkTo` function
+
+### The problem 
+Let's assume that the user wants to navigate like this:
+
+```
+1. Settings_root - navigate > Profile
+2. Profile - UP > Settings_root
+3. Settings_root - navigate > About 
+4. About - browser back button > Settings_root
+```
+
+Without minimal action, expected behavior won't be achieved and the final screen will be `Profile`.
+
+Broken behavior is the outcome of two things:
+1. Back button in the browser resets the navigation state with the state saved in step two.
+   
+2. `Navigation.navigate` creates action with `getActionFromState` dispatched high in the hierarchy. 
+
+The reason why `getActionFromState` provided by `react-navigation` is dispatched high in hierarchy is that it doesn't know about current navigation state, only about desired one.
+
+In this example it doesn't know if the `RightModalNavigator` and `Settings` are already mounted.
+
+
+The action for the first step looks like that: 
+```json
+{
+    "type": "NAVIGATE",
+    "payload": {
+        "name": "RightModalNavigator",
+        "params": {
+            "initial": true,
+            "screen": "Settings",
+            "params": {
+                "initial": true,
+                "screen": "Profile",
+                "path": "/settings/profile"
+            }
+        }
+    }
+}
+```
+
+
+That means, the params for the `RightModalNavigator` and `Settings` (also a navigator) will be filled with the information that we want to navigate to the `Profile` screen.
```

Comment:
Could you specify which part contains the information about wanting to navigate to `Profile`? The `path` param?